xds/src/test/java/io/grpc/xds/LoadBalancerConfigFactoryTest.java

```diff
@@ -322,13 +238,69 @@ public void maxRecursion() {
                                               buildWrrPolicy(
                                                 ROUND_ROBIN_POLICY))))))))))))))))))).build();
 
+    assertResourceInvalidExceptionThrown(cluster, false,
+        "Maximum LB config recursion depth reached");
+  }
+
+  private Cluster newCluster(Policy... policies) {
+    return Cluster.newBuilder().setLoadBalancingPolicy(
+        LoadBalancingPolicy.newBuilder().addAllPolicies(Lists.newArrayList(policies))).build();
+  }
+
+  private static Policy buildWrrPolicy(Policy... childPolicy) {
+    return Policy.newBuilder().setTypedExtensionConfig(TypedExtensionConfig.newBuilder()
+        .setTypedConfig(Any.pack(WrrLocality.newBuilder().setEndpointPickingPolicy(
+                LoadBalancingPolicy.newBuilder().addAllPolicies(Lists.newArrayList(childPolicy)))
+            .build()))).build();
+  }
+
+  private LbConfig newLbConfig(Cluster cluster, boolean enableLeastRequest)
+      throws ResourceInvalidException {
+    return ServiceConfigUtil.unwrapLoadBalancingConfig(
+        LoadBalancerConfigFactory.newConfig(cluster, enableLeastRequest));
+  }
+
+  private void assertValidRoundRobin(LbConfig lbConfig) {
+    assertThat(lbConfig.getPolicyName()).isEqualTo("wrr_locality_experimental");
+
+    @SuppressWarnings("unchecked") List<LbConfig> childConfigs = ServiceConfigUtil.unwrapLoadBalancingConfigList(
+        (List<Map<String, ?>>) lbConfig.getRawConfigValue().get("childPolicy"));
```

Comment:
I see this is pre-existing, but it was one of the places `JsonUtil.getListOfObjects()` could be used and was missed.

Reply:
Done.